### PR TITLE
Changed: Use EnterCommand's afterExecute event to alter it's behavior.

### LINKED ISSUE #28: EnterCommand at the end of heading should create paragraph.
This behavior already was implemented, but due to changes in `EnterCommand` now it is a special case and needs additional code to be handled.

### PR DESCRIPTION
**Pull request by scofalik**
Fixes #28

### CODE REVIEW

File: src/headingscommand.js [hunk 1]
```diff
@@ -64,11 +64,14 @@ export default class HeadingsCommand extends Command {
...
 	/**
-	 * Executes the command if it is enabled.
+	 * Executes command.
 	 *
+	 * @protected
 	 * @param {String} [formatId] The identifier of the heading format that should be applied. It should be one of the
 	 * {@link headings.HeadingsFormat heading formats} provided to the command constructor. If this parameter is not provided,
 	 * the value from {@link headings.HeadingsCommand#defaultFormat defaultFormat} will be used.
+	 * @returns {Object} data Data object, available in {@link core.Command#event:afterExecute}
+	 * @returns {engine.model.Batch} data.batch Batch created and used by the command.
```
Comment (Reinmar) [inline src/headingscommand.js]: Leftover?

File: tests/headingscommand.js [hunk 2]
```diff
@@ -53,6 +53,15 @@ describe( 'HeadingsCommand', () => {
 				expect( command.value ).to.equal( format );
 			} );
 		}
+
+		it( 'should be equal to defaultFormat if format has not been found', () => {
```
Comment (Reinmar) [inline tests/headingscommand.js]: Why can't it be null? Why is this change needed now?

Comment (scofalik) [inline tests/headingscommand.js]: This is because when you focus selection inside a model element that is not in `formats` array, you get this error:
`Uncaught TypeError: Cannot read property 'label' of undefined`.

This is because no option has been found to be displayed in the dropdown.

However, as I think of it now, it should probably be done like it is in CKE4. There is "Format" option there whenever you are inside something that is not recognized by `Headings` feature. Right now, in CKE5, you will have "Paragraph" option displayed. Because of that you are unable to actually change to the paragraph using `Headings` dropdown.

I think this is worthy of seperate issue because then we have to ask ourselves what is the `defaultValue` -- still `paragraph` or maybe "format"?

File: src/headingscommand.js (hunk 1 above)
Comment (scofalik) [inline src/headingscommand.js]: Yes, I fixed it.

File: tests/headingscommand.js (hunk 2 above)
Comment (Reinmar) [inline tests/headingscommand.js]: OK, so I'm fine with defaulting to the default format in such a case. But next time please do not squeeze such changes without mentioning them ;)

Comment (scofalik) [inline tests/headingscommand.js]: I am sorry, this is just [the way I are](https://www.youtube.com/watch?v=U5rLz5AZBIA).
